mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/memory/MemoryDeletedMessagesVault.java

```diff
@@ -34,18 +40,25 @@
 
 public class MemoryDeletedMessagesVault implements DeletedMessageVault {
     private final Table<User, MessageId, DeletedMessage> table;
+    private final Table<User, MessageId, byte[]> contentTable;
 
     MemoryDeletedMessagesVault() {
         table = HashBasedTable.create();
+        contentTable = HashBasedTable.create();
     }
 
     @Override
-    public synchronized Mono<Void> append(User user, DeletedMessage deletedMessage) {
+    public synchronized Mono<Void> append(User user, DeletedMessage deletedMessage, InputStream inputStream) {
```

Comment:
it's usually better to lock (synchronize) on a private object. With the current code, anybody putting a lock on this object will screw your locking logic.